src/applications/personalization/profile-2/components/ProfileWrapper.jsx

```diff
@@ -1,88 +1,123 @@
-import React, { useEffect } from 'react';
+import React, { Component } from 'react';
 import { connect } from 'react-redux';
 import LoadingIndicator from '@department-of-veterans-affairs/formation-react/LoadingIndicator';
 
 import RequiredLoginView from 'platform/user/authorization/components/RequiredLoginView';
 import backendServices from 'platform/user/profile/constants/backendServices';
+import {
+  createIsServiceAvailableSelector,
+  isMultifactorEnabled,
+} from 'platform/user/selectors';
 
 import {
   fetchMilitaryInformation as fetchMilitaryInformationAction,
   fetchHero as fetchHeroAction,
+  fetchPersonalInformation as fetchPersonalInformationAction,
 } from 'applications/personalization/profile360/actions';
+import { fetchPaymentInformation as fetchPaymentInformationAction } from 'applications/personalization/profile360/actions/paymentInformation';
 
 import ProfileHeader from './ProfileHeader';
 import ProfileSideNav from './ProfileSideNav';
 
-const ProfileWrapper = ({
-  children,
-  fetchFullName,
-  fetchMilitaryInformation,
-  showLoader,
-  user,
-}) => {
-  useEffect(
-    () => {
-      fetchMilitaryInformation();
-      fetchFullName();
-    },
-    [fetchMilitaryInformation, fetchFullName],
-  );
+class ProfileWrapper extends Component {
+  componentDidMount() {
+    const {
+      fetchFullName,
+      fetchMilitaryInformation,
+      fetchPersonalInformation,
+      fetchPaymentInformation,
+      shouldFetchDirectDepositInformation,
+    } = this.props;
+    fetchMilitaryInformation();
+    fetchFullName();
+    fetchPersonalInformation();
+    if (shouldFetchDirectDepositInformation) {
+      fetchPaymentInformation();
```

Comment:
we could probably extract all of this into a thunk to reduce the lines in the component, but its fine either way.

Reply:
I'd love to see an example of how that'd work, actually. I had this vague idea that I'd use a single thunk to make all these calls. But since we had all these actions already set up I figured I'd leverage what we had. But if you've got a better, cleaner way I'm totally open to it.

Reply:
```const fetchAllTheThings = shouldFetchDirectDepositInformation = async dispatch => {
  await fetchFullName();
  await fetchPersonalInformation();
   if (shouldFetchDirectDepositInformation) {
    await fetchPaymentInformation();
  }
}```

then in the component just invoke it
```
fetchAllTheThings(shouldFetchDirectDepositInformation)
```

Reply:
pseudo-y code, but should work

Reply:
I'm not as comfortable with async/await as I should be, but wouldn't that make each fetch call sequentially rather than in parallel? And if so, could you make the `fetch...` calls without `await`?

Reply:
I think you'd have to use `Promise.all` -

```js
const requests = [
   fetchFullName(),
   fetchPersonalInformation()
]

if (shouldFetchDirectDepositInformation) {
   requests.push(fetchPaymentInformation());
}

await Promise.all(requests);

```

The only catch is that if _any_ request fails then an error is thrown. But I think you'd want to render each component as they receive their data so that the whole profile isn't blocked if one request is slow.

I like the way this component is arranged using static chunks of JSX btw!

Reply:
we actually do want to block the entire profile until all data calls have resolved/failed. at least we don't want to render the new sidenav until all the calls hav resolved/failed. This is because the entries in the sidenav will depend on what data is available; ie, it's possible we won't show the Military History or Direct Deposit items in the sidenav.